docs/hello_nextflow/03_hello_plumbing.md

```diff
@@ -0,0 +1,755 @@
+
+In the workflow block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="56"
+    // convert the greeting to uppercase
+    convertToUpper()
+}
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="56"
+    // convert the greeting to uppercase
+    convertToUpper(sayHello.out)
+}
+```
+
+For a simple case like this (one output to one input), that's all we need to do to connect two processes!
+
+### 1.4. Run the workflow again with `-resume`
+
+Let's run this using the `-resume` flag, since we've already run the first step of the workflow successfully.
+
+```bash
+nextflow run hello-plumbing.nf -resume
+```
+
+There is now an extra line in the console output, which corresponds to the new process we just added:
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+Launching `hello-plumbing.nf` [cheeky_hamilton] DSL2 - revision: f7fbe8e223
+
+executor >  local (3)
+[45/eb4757] sayHello (2)       [100%] 3 of 3, cached: 3 ✔
+[ae/4579ab] convertToUpper (3) [100%] 3 of 3 ✔
+```
+
+Have a look inside the work directory of one of the calls to the second process.
+
+```bash
+tree -a work/ae/4579ab5b4f2c1d986d3a955e31f2b7/
+```
+
+You should find two output files listed: the output of the first process, and the output of the second.
+
+```console title="Output"
+work/ae/4579ab5b4f2c1d986d3a955e31f2b7/
+├── Holà-output.txt -> /workspace/gitpod/hello-nextflow/work/dc/93eab52bd47ef198b1cfe1a7721b4b/Holà-output.txt
+└── UPPER-Holà-output.txt
+```
+
+The output of the first process is in there because Nextflow staged it there in order to have everything needed for execution within the same subdirectory.
+However, it is actually a symbolic link pointing to the the original file in the subdirectory of the first process call.
+By default, Nextflow uses symbolic links rather than copies to stage input and intermediate files.
+
+You'll also find the final outputs in the `results` directory since we used the `publishDir` directive in the second process too.
+
+!!! note
+
+    All we did was connect the output of `sayHello` to the input of `convertToUpper` and the two processes could be run serially.
+    Nextflow did the hard work of handling individual input and output files and passing them between the two commands for us.
+    This is the power of channels in Nextflow, doing the busywork of connecting our pipeline steps together.
+
+### Takeaway
+
+You know how to add a second step that takes the output of the first step as input.
+
+### What's next?
+
+Learn how to collect outputs from batched process calls and feed them into a single process.
+
+---
+
+## 2. Add a third step to collect all the greetings
+
+When we apply a transformation to a batch of inputs, like we're doing here to the multiple greetings, we'll often want to collect the transformed outputs and feed them into a single step that performs some kind of analysis or summation.
```

Comment:
```suggestion
When we use a process to apply a transformation to each of the elements in a channel, like we're doing here to the multiple greetings, we sometimes want to collect elements from the output channel of that process, and feed them into another process that performs some kind of analysis or summation.
```

I think it's worth being specific with the channel/ process language, since we've introduced those concepts, so that people get used to them.